pkg/router/template/plugin.go

```diff
@@ -156,6 +156,10 @@ func (p *TemplatePlugin) HandleEndpoints(eventType watch.EventType, endpoints *k
 		if commit {
 			p.Router.Commit()
 		}
+	case watch.Deleted:
+		glog.V(4).Infof("Deleting endpoints for %s", key)
+		p.Router.DeleteEndpoints(key)
+		p.Router.Commit()
```

Comment:
The server lines for the endpoints are deleted from haproxy.config when the above is added.
DeleteEndpoints(key) actually does delete the endpoints. It does not get called anywhere else in the code.